mitodl/open-discussions
# Show most relevant availability for search result

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2162

#### What's this PR do?
- Sends the selected availability facets (if any) to the `LearningResourceCard` card component
- Adds functions for determining the best run/availability to display

#### How should this be manually tested?
- Select 'Available Now' facet only, all the results should show 'Available Now'
- Select 'In the next year' only, results should show a mix from 'In the next week' to 'In the next year'
- Select 'Available Now' and 'In the next year', results should show a mix from 'Available Now' to 'In the next year'.  
- Select 'Programs' and 'Learning Paths' facets, they should all be 'Available Now'

#### Screenshots

<img width="1589" alt="Screen Shot 2019-09-03 at 4 36 21 PM" src="https://user-images.githubusercontent.com/187676/64206876-71f0b880-ce69-11e9-9f13-e5048cb8df14.png">

<img width="1594" alt="Screen Shot 2019-09-03 at 4 37 10 PM" src="https://user-images.githubusercontent.com/187676/64206872-6e5d3180-ce69-11e9-9c81-36fb8c2c4ecc.png">
